File: data/python/tests/test_itertools_cycle.py
import unittest
from function_itertools_cycle import CycleProcessor

class TestCycleExample(unittest.TestCase):

    def setUp(self):
        self.processor = CycleProcessor()

    def test_cycle_example_empty_sequence(self):
        result = self.processor.test_cycle('', 5)
        self.assertEqual(result, '')

    def test_cycle_example_single_character(self):
        result = self.processor.test_cycle('A', 5)
        self.assertEqual(result, 'AAAAA')

    def test_cycle_example_short_sequence(self):
        result = self.processor.test_cycle('AB', 5)
        self.assertEqual(result, 'ABABA')

    def test_cycle_example_long_sequence(self):
        result = self.processor.test_cycle('ABCDEFG', 10)
        self.assertEqual(result, 'ABCDEFGABC')

    def test_cycle_example_numeric_sequence(self):
        result = self.processor.test_cycle('123', 7)
        self.assertEqual(result, '1231231')

if __name__ == '__main__':
    unittest.main()

File: manual/data/java/tests/test_itertools_cycle.java
package com.example;

import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class test_itertools_cycle {
    private function_itertools_cycle processor;

    @BeforeEach
    public void setUp() {
        processor = new function_itertools_cycle();
    }

    @Test
    public void test_cycle_example_empty_sequence() {
        String result = processor.test_cycle("", 5);
        assertEquals("", result);
    }

    @Test
    public void test_cycle_example_single_character() {
        String result = processor.test_cycle("A", 5);
        assertEquals("AAAAA", result);
    }

    @Test
    public void test_cycle_example_short_sequence() {
        String result = processor.test_cycle("AB", 5);
        assertEquals("ABABA", result);
    }

    @Test
    public void test_cycle_example_long_sequence() {
        String result = processor.test_cycle("ABCDEFG", 10);
        assertEquals("ABCDEFGABC", result);
    }

    @Test
    public void test_cycle_example_numeric_sequence() {
        String result = processor.test_cycle("123", 7);
        assertEquals("1231231", result);
    }
}